docs/dotnet/dotnet-kestrel-metrics.md

```diff
@@ -0,0 +1,221 @@
+|---|---|---|---|---|
+| `kestrel.endpoint` | string | TODO [1] | `TODO` | Recommended |
+
+**[1]:** **TODO: why not server.**? What's KEstrel-specific about it**
+<!-- endsemconv -->
+
+## Metric: `kestrel.connection.duration`
+
+**Status**: [Experimental][DocumentStatus]
+
+This metric is required.
+
+<!-- semconv metric.kestrel.connection.duration(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    |
+| -------- | --------------- | ----------- | -------------- |
+| `kestrel.connection.duration` | Histogram | `s` | The duration of connections on the server. [1] |
+
+**[1]:** Meter name is `Microsoft.AspNetCore.Server.Kestrel`
+<!-- endsemconv -->
+
+<!-- semconv metric.kestrel.connection.duration(full) -->
+| Attribute  | Type | Description  | Examples  | Requirement Level |
+|---|---|---|---|---|
+| `exception.type` | string | The type of the exception (its fully-qualified class name, if applicable). The dynamic type of the exception should be preferred over the static type in languages that support it. [1] | `java.net.ConnectException`; `OSError` | Recommended |
+| `kestrel.endpoint` | string | TODO [2] | `TODO` | Recommended |
+| [`network.protocol.version`](../general/attributes.md) | string | Version of the application layer protocol used. See note below. [3] | `3.1.1` | Recommended |
+| `tls.protocol` | string | TODO [4] | `TODO` | Recommended |
+| [`url.scheme`](../url/url.md) | string | The [URI scheme](https://www.rfc-editor.org/rfc/rfc3986#section-3.1) component identifying the used protocol. | `https`; `ftp`; `telnet` | Recommended |
+
+**[1]:** **TODO: Let's define common attribute for error reason/status**
+
+**[2]:** **TODO: why not server.**? What's KEstrel-specific about it**
+
+**[3]:** `network.protocol.version` refers to the version of the protocol used and might be different from the protocol client's version. If the HTTP client used has a version of `0.27.2`, but sends HTTP version `1.1`, this attribute should be set to `1.1`.
+
+**[4]:** **TODO: why not network.protocol.***?
+<!-- endsemconv -->
+
+## Metric: `kestrel.rejected_connections`
+
+**Status**: [Experimental][DocumentStatus]
+
+This metric is required.
+
+<!-- semconv metric.kestrel.rejected_connections(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    |
+| -------- | --------------- | ----------- | -------------- |
+| `kestrel.rejected_connections` | Counter | `{connection}` | Number of connections rejected by the server. [1] |
+
+**[1]:** Connections are rejected when the currently active count exceeds the value configured with MaxConcurrentConnections.
+Meter name is `Microsoft.AspNetCore.Server.Kestrel`
+
+**TODO: do we ever will have other rejection reasons**
+<!-- endsemconv -->
+
+<!-- semconv metric.kestrel.rejected_connections(full) -->
+| Attribute  | Type | Description  | Examples  | Requirement Level |
+|---|---|---|---|---|
+| `kestrel.endpoint` | string | TODO [1] | `TODO` | Recommended |
```

Comment:
To be clear, if a user specified the endpoint then endpoint.ToString() could be anything they wrote without even being aware the string would show up like this. It might be some mixture of transport information, protocol information, address information, and unrelated information all mushed into one opaque string. So a made up example - the attribute could look like: network.transport = "Address:0x12385da93f08000 Size:0x60000 Debug:False ACL:ReadWrite(joe@contuso.org)"

I'm OK with that but I assume it would be rare and the folks impacted are probably closely related to those who wrote the code, but just calling it out :)